dashboard/config/routes.rb

```diff
@@ -317,6 +317,13 @@ def redirect_to_teacher_dashboard
 
   post '/plc/course_units/:id/submit_new_questions_and_answers', to: 'plc/course_units#submit_new_questions_and_answers'
 
+  namespace :pd do
+    get 'workshops/:workshop_id/enroll', action: 'new', controller: 'workshop_enrollment'
+    post 'workshops/:workshop_id/enroll', action: 'create', controller: 'workshop_enrollment'
+    get 'workshop_enrollment/:code', action: 'show', controller: 'workshop_enrollment'
```

Comment:
`workshops/x/enroll` creates a new enrollment for workshop x. In order not to easily discover or accidentally cancel other people's enrollments I didn't want enrollments to be accessed by an incremental id like `workshops/x/enrollments/y` so I used a random code. When recalling the enrollment by code, the workshop id is not needed so `workshops/x/enrollments/:code` seemed unnecessary. However, maybe that's cleaner?

Alternatively it could be `/workshops/enrollments/:code`, or just `enrollments/:code`. What do you think?